src/game/SpellEffects.cpp

```diff
@@ -1149,6 +1149,13 @@ void Spell::EffectDummy(SpellEffectIndex eff_idx)
 
                     return;
                 }
+                case 32027:                                 // Expedition Flare
+                {
+                    if (!m_CastItem)                        // 32029 = Expedition Preserver | 32030 = Expedition Scout
```

Comment:
Well, then you wouldnt wanna send the m_castItem along at all, because it will expire every time. It depends on how you look at cast items, either A) spell is tied to castItem or B) castItem is tied to spell, when its a key unlock spell i assume the former, while when its some script spell, I assume the latter. This cas is in my opinion the latter.